python/artm/scores.py

```diff
@@ -0,0 +1,515 @@
+        return self._config
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def class_id(self):
+        return self._class_id
+
+    @property
+    def topic_names(self):
+        return self._topic_names
+
+    @property
+    def model(self):
+        return self._model
+
+    @property
+    def score(self):
+        return self._score
+
+    @property
+    def master(self):
+        return self._master
+
+    @class_id.setter
+    def class_id(self, class_id):
+        _reconfigure_field(self, class_id, 'class_id')
+
+    @topic_names.setter
+    def topic_names(self, topic_names):
+        _reconfigure_field(self, topic_names, 'topic_names')
+
+
+###################################################################################################
+# SECTION OF SCORE CLASSES
+###################################################################################################
+class SparsityPhiScore(BaseScore):
+    """SparsityPhiScore is a score in ArtmModel (public class)
+
+    Args:
+      name (str): the identifier of score, will be auto-generated if not specified
+      class_id (str): class_id to score, default=None
+      topic_names (list of str): list of names of topics to regularize, will
+      score all topics if not specified
+      eps (double): the tolerance const, everything < eps
+      considered to be zero, default=1e-37
+    """
+
+    _type = lib.ScoreConfig_Type_SparsityPhi
+
+    def __init__(self, name=None, class_id=None, topic_names=None, eps=None):
+        BaseScore.__init__(self,
+                           name=name,
+                           class_id=class_id,
+                           topic_names=topic_names)
+        self._eps = GLOB_EPS
+
+        if eps is not None:
+            self._config.eps = eps
+            self._eps = eps
+
+    @property
+    def eps(self):
+        return self._eps
+
+    @eps.setter
+    def eps(self, eps):
+        _reconfigure_field(self, eps, 'eps')
+
+
+class SparsityThetaScore(BaseScore):
+    """SparsityThetaScore is a score in ArtmModel (public class)
+
+    Args:
+      name (str): the identifier of score, will be auto-generated if not specified
+      topic_names (list of str): list of names of topics to regularize, will
+      score all topics if not specified
+      eps (double): the tolerance const, everything < eps
+      considered to be zero, default=1e-37
+    """
+
+    _type = lib.ScoreConfig_Type_SparsityTheta
+
+    def __init__(self, name=None, topic_names=None, eps=None):
+        BaseScore.__init__(self,
+                           name=name,
+                           class_id=None,
+                           topic_names=topic_names)
+        self._eps = GLOB_EPS
+
+        if eps is not None:
+            self._config.eps = eps
+            self._eps = eps
+
+    @property
+    def eps(self):
+        return self._eps
+
+    @property
+    def class_id(self):
+        raise KeyError('No class_id parameter')
```

Comment:
Very strange. Are you defined this property in the base class to disable it in the successor?